src/mammos_ai/_beyond_stoner_wohlfarth_fixed_angle/__init__.py

```diff
@@ -0,0 +1,232 @@
+            # array with shape (n_samples_in_class, 3) containing
+            # [Hc, Mr, BHmax] predictions
+            res = session.run(None, {session.get_inputs()[0].name: X_subset})[0]
+            y_log[mask] = res
+
+    # 5. Postprocess
+    y = np.expm1(y_log)
+
+    # Reshape output as y.shape = (n_samples_in_class, 3)
+    return y[0] if is_scalar else y.reshape(original_shape + (3,))
+
+
+def Hc_Mr_BHmax_from_Ms_A_K(
+    Ms: mammos_entity.Entity | astropy.units.Quantity | numbers.Number | np.ndarray,
+    A: mammos_entity.Entity | astropy.units.Quantity | numbers.Number | np.ndarray,
+    K1: mammos_entity.Entity | astropy.units.Quantity | numbers.Number | np.ndarray,
+    model: str = "cube50_singlegrain_random_forest_v0.1",
+) -> mammos_analysis.hysteresis.ExtrinsicProperties:
+    """Predict Hc, Mr and BHmax from micromagnetic properties Ms, A and K1.
+
+    This function predicts extrinsic properties coercive field Hc, remanent
+    magnetization Mr and maximum energy product BHmax given a set of micromagnetic
+    material parameters.
+
+    Different models are available for the prediction. The following list provides an
+    overview:
+
+    - ``cube50_singlegrain_random_forest_v0.1``: Random forest model trained on
+      simulated data for single grain cubic particles with 50 nm edge length with
+      the external field applied parallel to the anisotropy axis. These are both
+      aligned along an edge of the cube. Further details on the training data
+      and model can be found in the
+      `model repository <https://github.com/MaMMoS-project/ML-models/tree/main/beyond-stoner-wohlfarth/single-grain-easy-axis-model>`_.
+
+
+
+    Args:
+       Ms: Spontaneous magnetization.
+       A: Exchange stiffness constant.
+       K1: Uniaxial anisotropy constant.
+       model: AI model used for the prediction
+
+    Returns:
+       An object containing extrinsic properties Hc, Mr, BHmax
+
+    Examples:
+    >>> import mammos_ai
+    >>> import mammos_entity as me
+    >>> mammos_ai.Hc_Mr_BHmax_from_Ms_A_K(me.Ms(1e6), me.A(1e-12), me.Ku(1e6))
+    ExtrinsicProperties(Hc=..., Mr=..., BHmax=...)
+    """
+    Ms = me.Ms(Ms, unit=u.A / u.m)
+    A = me.A(A, unit=u.J / u.m)
+    K1 = me.Ku(K1, unit=u.J / u.m**3)
+
+    Ms_arr = np.atleast_1d(Ms.value)
+    A_arr = np.atleast_1d(A.value)
+    K1_arr = np.atleast_1d(K1.value)
+
+    if not (Ms_arr.shape == A_arr.shape == K1_arr.shape):
+        raise ValueError(
+            f"Input arrays must have the same shape. Shapes are Ms: {Ms_arr.shape}, "
+            f"A: {A_arr.shape}, Ku: {K1_arr.shape}"
+        )
+
+    original_shape = Ms_arr.shape
+    is_scalar = Ms_arr.ndim == 0 or (Ms_arr.ndim == 1 and Ms_arr.size == 1)
+
+    match model:
+        case "cube50_singlegrain_random_forest_v0.1":
+            y = _predict_cube50_singlegrain_random_forest_v0_1(
+                Ms, A, K1, Ms_arr, A_arr, K1_arr, original_shape, is_scalar
```

Comment:
I don't really like that we pass the entities and the arrays. Maybe we could either just pass the entities or run the classification first and pass the classification result to this function?